File: mergeSort.js
function printArray(arr) {
    for (let num of arr) {
        process.stdout.write(num + " ");
    }
    console.log();
}

function merge(arr, mid, low, high) {
    let i = low;
    let j = mid + 1;
    let k = low;
    let B = [];

    while (i <= mid && j <= high) {
        if (arr[i] < arr[j]) {
            B[k] = arr[i];
            i++;
            k++;
        } else {
            B[k] = arr[j];
            j++;
            k++;
        }
    }

    while (i <= mid) {
        B[k] = arr[i];
        k++;
        i++;
    }

    while (j <= high) {
        B[k] = arr[j];
        k++;
        j++;
    }

    for (let i = low; i <= high; i++) {
        arr[i] = B[i];
    }
}

function mergeSort(arr, low, high) {
    if (low < high) {
        let mid = Math.floor((low + high) / 2);
        mergeSort(arr, low, mid);
        mergeSort(arr, mid + 1, high);
        merge(arr, mid, low, high);
    }
}

function main() {
    const readline = require('readline');
    const rl = readline.createInterface({
        input: process.stdin,
        output: process.stdout
    });

    rl.question("Enter the number of elements: ", (n) => {
        let arr = [];

        console.log("Enter the elements of the array:");
        rl.on('line', (input) => {
            arr.push(parseInt(input));
            if (arr.length === parseInt(n)) {
                console.log("Array before sorting:");
                printArray(arr);

                mergeSort(arr, 0, n - 1);

                console.log("Array after sorting:");
                printArray(arr);
                rl.close();
            }
        });
    });
}

main();

File: mergeSort.c
#include<stdio.h>

void printArray(int *arr, int n)
{
    for (int i = 0; i<n; i++)
    {
        printf("%d ", arr[i]);
    }

    printf("\n");
}
void merge(int arr[], int mid,int low,int high)
{
    int i,j, k, B[100];
    i = low;
    j = mid + 1;
    k = low;

    while (i <=mid && j <= high)
    {
      if (arr[i] < arr[j])
      {
        B[k] = arr[i];
        i++;
        k++;

      }
      else
      {
        B[k] = arr[j];
        j++;
        k++;

      }

    }
    while(i<= mid)
    {
        B[k] = arr[i];
        k++;
        i++;
    }

    while(j <=high)
    {
        B[k] = arr[j];
        k++;
        j++;
    }
    for(int i=low; i<=high; i++)
    {
        arr[i] = B[i];
    }

}

    void mergeSort(int arr[], int low, int high)
    {
        int mid;
        if(low<high)

        {
            mid = (low + high)/2;
            mergeSort(arr, low, mid);
            mergeSort(arr, mid + 1, high);
            merge(arr, mid, low, high);
        }
    }



int main()
{
    int n,i; 
    printf("Enter the number of elements: ");
    scanf("%d", &n);

    int arr[n];
    printf("Enter the elements of array");
    for( int i = 0; i <n; i++)
    {

        scanf("%d", &arr[i]);
    }

    printArray(arr, n);
    mergeSort(arr,0 ,n) ;
    printArray(arr, n);
    return 0;


}